qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/IndexingIT.java

```diff
@@ -410,19 +409,29 @@ private void assertCount(String index, int count) throws IOException {
         );
     }
 
-    private Version minNodeVersion() throws IOException {
+    private boolean nodeSupportBulkApi(Map<?, ?> nodeInfo) {
+        // TODO[lor]: replace this check with a (historical) feature check ("supports bulk requests")
+        var versionString = nodeInfo.get("version").toString();
+        Version version;
+        try {
+            version = Version.fromString(versionString.replace("-SNAPSHOT", ""));
+        } catch (IllegalArgumentException ignored) {
```

Comment:
When would this case happen that we would need to fall back?

Reply:
This shouldn't happen in serverless because we do not replace the version string (it is baked into the build). It is still the on-prem semantic version, for now. I think we should let any failure to parse the version propagate to fail the test.

Reply:
On this one I disagree: the version string in serverless is already the build hash, and very soon these rest tests that get the version from `GET _nodes` like this one will receive the build hash in the "version" field.
I agree on not relying on exceptions, but we need to differentiate semantic (parseable) versions and opaque versions now (e.g. via a regex as we did elsewhere). We can (for now) assume safely non-semantic versions are serverless and have the bulk API.
In the future we will replace this with a check on features, but we need a change in the rest API for that, and it may not happen so soon. I'd rather have a fallback for now.